WordPress/gutenberg
# Data views: Add click-to-select behavior on table rows

## What?
Capture clicks on on each table row and select the associated record accordingly.

## Why?
A larger 'hitbox' makes it a bit easier for mouse / trackpad users to select items.

## How?
Adjusts the keyboard listener we had to include clicks.

## Testing Instructions
1. Open a data view like Manage all pages and switch to table layout.
2. Observe that clicking a row will select the item.


https://github.com/WordPress/gutenberg/assets/846565/634172af-146e-4713-8c51-ccf9b255e04b



### Notes:

* It might be nice to follow-up with shift+click and cmd+click multiselection.
